packages/components/src/components/step/step.ts

```diff
@@ -0,0 +1,176 @@
+import { css, unsafeCSS } from 'lit';
+import { customElement } from '../../internal/register-custom-element';
+import { html, literal } from 'lit/static-html.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { property } from 'lit/decorators.js';
+import componentStyles from '../../styles/component.styles';
+import cx from 'classix';
+import ParagraphStyles from '../../styles/paragraph/paragraph.css?inline';
+import SolidElement from '../../internal/solid-element';
+/**
+ * @summary Steps are used inside [step groups](/components/step-group) to guide users through the steps of a process or task..
+ * @documentation https://solid.union-investment.com/[storybook-link]/step
+ * @status stable
+ * @since 3.2.0
+ *
+ * @dependency sd-icon
+ * @dependency sd-divider
+ *
+ * @slot - The step's description.
+ * @slot label - The step's label.
+ *
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart circle-and-tail-container - The container that wraps the step's circle and tail.
+ * @csspart circle - The circle that marks the step's state.
+ * @csspart tail - The step's tail.
+ * @csspart text-container - The container that wraps the step's label and description.
+ * @csspart label - The step's label.
+ * @csspart description - The step's description.
+ */
+@customElement('sd-step')
+export default class SdStep extends SolidElement {
+  /** The step's size. */
+  @property({ reflect: true }) size: 'lg' | 'sm' = 'lg';
+
+  /** Determines the orientation of the step. */
+  @property({ reflect: true }) orientation: 'horizontal' | 'vertical' = 'horizontal';
+
+  /** Determines the state of the step. */
+  @property({ reflect: true }) state: 'waiting' | 'inProgress' | 'finished' = 'waiting';
```

Comment:
I don't see the need to change it to something less component specific as the component will only be used for this context. If we want to change it to something with less context (which I would be totally fine with), we need to change it in design too and I would then go further to something like: 
- past
- current
- upcoming

Reply:
I personally don't mind the states being component specific, if they describe the role more accurately. I hesitate to use 'disabled' for example as that provides no information about the positioning of the step, while 'current' and 'complete' imply an order. 

I would prefer the original state names we used. Thoughts? @mariohamann @karlbaumhauer

Reply:
@mariohamann what would be the reason to have it non-component-specific? Do you see the component being used for something else but classic steps? 

Mayebe we should clarify this first before we try to find a wording which sounds better... 🤣 @coraliefeil / @EllaELSA any thoughts on this?

Reply:
Thanks for all the thoughts, that helped me to have further thoughts on that. Let's take a step back:

Step groups in general can be `interactive` and `non-interactive` – if they are `non-interactive`, they usually all have the same state, which is `blue`.

For the discussion here we should focus on `interactive` elements. Here the `step-group` serves as a navigation, having `role=navigation` (if there are interactive elements – @Vahid1919!). Inside that, we either have `links` or `buttons`. There are roles to influence the semantics of these items, which we use in 'sd-navigation-item`: `disabled` and `current`. The latter e. g. provides a specific attribute which in this case should be `aria-current="step"` (https://www.accede-web.com/en/guidelines/html-css/complementary-guidelines/identify-the-current-step-in-multiple-steps-forms-using-aria-currentstep/).

Let's make a comparison here:

![CleanShot 2024-05-07 at 16 29 42](https://github.com/solid-design-system/solid/assets/26542182/c0596aa2-22a8-4d24-a821-ce156f7b444c)

Seeing this, taking a11y, semantics and consistency into account, and after our talk @Vahid1919, I therefore would propose:

- `disabled` → boolean to make it look disabled and make it semantically a disabled button
- `current` → boolean to make it behave look and from aria-roles `current`

If nothing is set, it's just behaving as "default".

Reply:
I think this makes especially sense if applications decide, that all elements in the steps should be clickable – which could totally be a case! Then everything besides `current` has the default state and there even is now `disabled` state. I don't see a reason, why this should be prevented @EllaELSA @coraliefeil? It could be problematic for forms, but if you think about interactive mini-apps like "Get to know more about process to buy funds by clicking on the steps" ore similar. (I hope this example is some understandable. 😆 )

Reply:
thx @mariohamann for the explanation. looking at it from a navigation poit of view and not from a steps perspective, I agree to the suggestion of default, current and disabled. 

I would just love to clarify your last question towards @EllaELSA (in terms of form elements) and in any case add this information to the documentation. Just from the naming og the component it sounds strange but with this information, it is more clear and also shows the generic structure of our components.